tokio/src/task/local.rs

```diff
@@ -280,10 +280,41 @@ pin_project! {
 
 tokio_thread_local!(static CURRENT: LocalData = const { LocalData {
     ctx: RcCell::new(),
+    entered: Cell::new(false),
 } });
 
 struct LocalData {
     ctx: RcCell<Context>,
+    entered: Cell<bool>,
+}
+
+impl LocalData {
+    /// Should be called except when we call `LocalSet::enter`.
+    /// Especially when we poll a LocalSet.
+    #[must_use = "dropping this guard will reset the entered state"]
+    fn enter(&self, ctx: Rc<Context>) -> LocalDataEnterGuard<'_> {
+        let ctx = self.ctx.replace(Some(ctx));
+        let entered = self.entered.replace(false);
+        LocalDataEnterGuard {
+            local_data_ref: self,
+            ctx,
+            entered,
```

Comment:
It's because of the name of the user-facing level functions. So this `LocalData::enter` should be called when we poll, and the value `entered` should be set only if user called the `LocalSer::enter`, which is public. 
I agree that it's confused, but could not find suitable naming..

Reply:
I agree with @Darksonn, the `entered` field could be renamed to something clearer.

Reply:
Agreed

Reply:
Thank you very much! I've renamed it!
But... `wake_on_spawn_local` vs `wake_on_schedule`.. which would be better?

Reply:
`wake_on_schedule` also makes sense to me.

Reply:
Thank you! I've renamed it to `wake_on_schdule`!